bundles/tampere/content-editor/view/SideContentEditor.js

```diff
@@ -1089,77 +1095,104 @@ Oskari.clazz.define('Oskari.tampere.bundle.content-editor.view.SideContentEditor
             if (me.layerGeometryType == "MultiLineString" || me.layerGeometryType == "GeometryPropertyType") {
                 lineButton.on('click', function() {
                         me.drawingActive = true;
-                        me.startNewDrawing({
-                            drawMode: 'line'
-                        });
+                        me.startNewDrawing();
                 });
             } else {
                 lineButton.addClass("disabled");
             }
 
-            var areaButton = $("<div />").addClass('add-area tool').attr('title', me.loc.tools.area);
+            var areaButton = jQuery("<div />").addClass('add-area tool').attr('title', me.loc.tools.area);
             if (me.layerGeometryType == "MultiPolygon" || me.layerGeometryType == "Polygon" || me.layerGeometryType == "GeometryPropertyType") {
                 areaButton.on('click', function() {
                         me.drawingActive = true;
-                        me.startNewDrawing({
-                            drawMode: 'area'
-                        });
+                        me.startNewDrawing();
                 });
             } else {
                 areaButton.addClass("disabled");
             }
 
-            var geomEditButton = $("<div />").addClass('selection-area tool').attr('title', me.loc.tools.geometryEdit);
+            var geomEditButton = jQuery("<div />").addClass('selection-area tool').attr('title', me.loc.tools.geometryEdit);
             if (me.layerGeometryType != null) {
 
                 geomEditButton.on('click', function() {
                     me.drawingActive = true;
                     me.drawToolType = "edit";
                     me.clickedGeometryNumber = null;
-                    if (me.layerGeometries.geometry.id.indexOf("Multi") > -1) {
-                        if (me.layerGeometries.geometry.components != undefined) {
-                            for (var i = 0; i < me.layerGeometries.geometry.components.length; i++) {
-                                if (me.layerGeometries.geometry.components[i].atPoint({lon:me.clickCoords.x,lat:me.clickCoords.y}, 2, 2))
-                                {
-                                    me.clickedGeometryNumber = i;
-                                    break;
+                    var geometryType = me.layerGeometries.geometry.getType();
+                    if (geometryType.indexOf("Multi") > -1) {
+                        if (me.layerGeometries.geometry.getCoordinates() != undefined) {
+                            var exists = false;
+                            for (var i = 0; i < me.layerGeometries.geometry.getCoordinates().length; i++) {
+                                if(geometryType === "MultiPoint") {
+                                    var closestPoint = me.layerGeometries.geometry.getClosestPoint([me.clickCoords.x, me.clickCoords.y]);
+                                    var distance = (
```

Comment:
This calculation doesn't work properly for geographic coordinates. However it might yield correct enough results for this purpose.